kola/tests/misc/selinux.go

```diff
@@ -27,42 +29,148 @@ func init() {
 		Flags:       []register.Flag{register.NoEnableSelinux},
 		Distros:     []string{"cl", "rhcos"},
 	})
+
+	register.Register(&register.Test{
+		Run:         SelinuxBoolean,
+		ClusterSize: 1,
+		Name:        "coreos.selinux.boolean",
+		Distros:     []string{"cl", "rhcos"},
+	})
+
+	register.Register(&register.Test{
+		Run:         SelinuxBooleanPersist,
+		ClusterSize: 1,
+		Name:        "coreos.selinux.boolean.persist",
+		Distros:     []string{"rhcos"},
+	})
+
+	register.Register(&register.Test{
+		Run:         SelinuxManage,
+		ClusterSize: 1,
+		Name:        "coreos.selinux.manage",
+		Distros:     []string{"rhcos"},
+	})
 }
 
-// SelinuxEnforce checks that some basic things work after `setenforce 1`
-func SelinuxEnforce(c cluster.TestCluster) {
-	m := c.Machines()[0]
+type cmdCheckOutput struct {
+	cmdline     string // command to run
+	checkoutput bool   // should output be checked
+	match       string // regex used to match output from command
+}
 
-	for _, cmd := range []struct {
-		cmdline     string
-		checkoutput bool
-		output      string
-	}{
-		{"getenforce", true, "Permissive"},
-		{"sudo setenforce 1", false, ""},
-		{"getenforce", true, "Enforcing"},
-		{"systemctl --no-pager is-active system.slice", true, "active"},
-		{"sudo cp --remove-destination $(readlink -f /etc/selinux/config) /etc/selinux/config", false, ""},
-		{"sudo sed -i 's/SELINUX=permissive/SELINUX=enforcing/' /etc/selinux/config", false, ""},
-	} {
+// testSelinuxCmds will run a list of commands, optionally check their output, and
+// ultimately reboot the host
+func testSelinuxCmds(c cluster.TestCluster, m platform.Machine, cmds []cmdCheckOutput) {
+	for _, cmd := range cmds {
 		output := c.MustSSH(m, cmd.cmdline)
 
-		if cmd.checkoutput && string(output) != cmd.output {
-			c.Fatalf("command %q has unexpected output: want %q got %q", cmd.cmdline, cmd.output, string(output))
+		if cmd.checkoutput {
+			match := regexp.MustCompile(cmd.match).MatchString(string(output))
+			if !match {
+				c.Fatalf("command %q has unexpected output: tried to match regexp %q, output was %q", cmd.cmdline, cmd.match, string(output))
+
+			}
 		}
 	}
 
 	err := m.Reboot()
 	if err != nil {
 		c.Fatalf("failed to reboot machine: %v", err)
 	}
+}
+
+// SelinuxEnforce checks that some basic things work after `setenforce 1`
+func SelinuxEnforce(c cluster.TestCluster) {
+	// we use the full path to most commands here because the $PATH is
+	// weird in RHCOS.  See https://github.com/openshift/os/issues/191
+	cmdList := []cmdCheckOutput{
+		{"/usr/sbin/getenforce", true, "Permissive"},
```

Comment:
Rather than changing the paths in the `SelinuxEnforce` function which is used by CL can we create an RHCOS variant that uses the full paths (like this) and then just drop the function & move back to the `SelinuxEnforce` method if/once the `PATH` variable is fixed on RHCOS?

Reply:
Yeah, I can do this.

It gets into a larger question of how much customization do we want to do to support RHCOS?  I was worried about cluttering up the code base with custom functions/tests that run only on RHCOS.  But if you think that would be OK, I'll start using that approach in other places.

Reply:
This is all from my perspective, @ajeddeloh or @bgilbert might have a differing one, but I'd prefer us to have the variant functions (even for single distributions). Especially in cases like this where it's very possible that we fix the underlying `PATH` issue and can then switch back to the existing test.